projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/monitoring/DataJobMonitorTest.java

```diff
@@ -532,6 +532,19 @@ void testRecordJobExecutionStatus_withNewExecutionIdAndNewStatus_shouldUpdateLas
         Assertions.assertEquals(jobExecution.getEndTime(), actualJob.get().getLastExecutionEndTime());
     }
 
+    @Test
+    @Order(31)
```

Comment:
The test relies on the previous test having set execution status already ... I feel it would have been better if we make the tests independent. It seems the ship has sailed for this class . 

Let's be careful with using order though. I think having dependencies between tests causes some issues - it makes them harder to understand. If one test fails all dependent tests may fail and this makes it harder to troubleshoot. It's also not possible to parallelize in the future.

Reply:
Correct, this test depends on the previous one. Also, I agree that dependent tests are hard to maintain and troubleshoot and will refrain from using them in the future.